src/Niv/Git/Cmd.hs

```diff
@@ -226,14 +245,29 @@ abortNoRev args l = abortGitBug args $ "Could not read revision from: " <> l
 abortNoRef :: [T.Text] -> T.Text -> IO a
 abortNoRef args l = abortGitBug args $ "Could not read reference from: " <> l
 
--- | Run the "git" executable
-runGit :: [T.Text] -> IO [T.Text]
-runGit args = do
+-- TODO: only clone shallow repository and fetch needed commits to speed up verification
+ensureAncestor :: T.Text -> T.Text -> T.Text -> IO T.Text
```

Comment:
I'm trying to wrap my head around what's happening here. If I understand `--rollback-protection` correctly, the goal is to prevent rolling back to an ancestor. So here you ensure that the _old_ rev is an ancestor of the _new_ rev.

Wouldn't that prevent a user from changing branches?

```
         o---o---o---B
        /
---o---o---o---A
```

If the user was on revision `A`, and changed the branch to `B`, wouldn't this implementation throw an error?

Would it make sense to invert the check and do something like this?

```haskell
ensureNotAncestor ... = 
  exitCode <- readProcessWithExitCode' "git" ["merge-base", "--is-ancestor", newRev, oldRev]
  case ExitCode
    ExitSuccess -> fail "new rev should not be an ancestor"
    _ -> pure ()
```

Reply:
Yes, it would prevent branch changes iff the new branch head is not an ancestor of the old rev.
I see that this might be surprising for users that 'just' want to switch branches.
Maybe an additional update flag `--ignore-rollback-protection` would make sense in these cases?

Could you give an example for when the inverted check would be sensible?
`ensureNotAncestor could for example succeed if I switched from a recent state to an old branch-off, but fail if the old rev was before the branching:

```
         o---o---o---C
        /
---A---o---o---o---B
```

```
niv update dep # A -> B
niv modify -b fork
niv update dep # B -> C, succeeds
```

```
niv modify -b fork
niv update dep # A -> C, fails
```